client/src/rr/dnssec/keypair.rs

```diff
@@ -241,11 +242,12 @@ impl KeyPair {
     /// # Return
     ///
     /// the KEY record data
+    #[allow(deprecated)]
     pub fn to_sig0key(&self, algorithm: Algorithm) -> DnsSecResult<KEY> {
         self.to_public_bytes()
             .map(|bytes| {
                      KEY::new(Default::default(),
-                              Default::default(),
+                              KeyUsage::Zone,
```

Comment:
I agree, this is confusing! But I think I came to a similar conclusion as you. In effect Zone became (was maintained for backward/cross compatibility) bit 7 in DNSKEY. My reading of the RFCs is that Zone is only valid in the context of a DNSKEY, and it became unclear to me why KEY would ever have Zone associated, when DNSKEY should really be used in that case.

Maybe I interpreted that incorrectly, but it seemed to be that KEY only became a valid record type for the storage of public key material, without any associated usage information. That is, usage itself should be defined through some other out-of-band means. For example, in the BIND config, an associated KEY needs to be added to the config as capable of performing DNS Update operations, for example:

https://github.com/bluejekyll/trust-dns/blob/master/compatibility/tests/conf/bind-example.conf#L49

All the deprecation between the various RFCs at this point in the DNSSec and SIG0 arena has created a unclear picture of the current state of the universe.